pytorch_lightning/utilities/migration.py

```diff
@@ -14,14 +14,10 @@
 from __future__ import annotations
 
 import sys
-import threading
 from types import ModuleType, TracebackType
 
 import pytorch_lightning.utilities.argparse
 
-# Create a global lock to ensure no race condition with deleting sys modules
-_lock = threading.Lock()
-
```

Comment:
@jdavid04 I don't see how removing the lock is solving the issue. Is this a bad merge?

Reply:
Yes sorry, I mentioned this in the original issue. I had additional issues with the thread locking in place when using Lightning with Ray Tune. Model selection never terminates and I believe it's related to the addition of this lock because it doesn't occur without it. I propose to remove it because the fix below should fix the multithreading issues with checkpointing also.

Reply:
@jdavid04 Ok. Unfortunately we added the thread locking here just a few days ago to solve another issue reported by users. cc @krshrimali 

We should try to find a fix for both.

Reply:
Could you explain to us (ideally with a reproducible example) what issues you are experiencing with the lock and why it is happening?

Reply:
Sure, I'll have a chance to get back to this in a few days and will try to reproduce both issues with a simpler example.